In [ ]:
# %matplotlib widget

from __future__ import annotations

import re
from collections import defaultdict
from dataclasses import dataclass, field
from itertools import permutations, product
from math import inf
from random import choice

import matplotlib.colors as mcolors
import matplotlib.pyplot as plt
import networkx as nx
import numpy as np
import numpy.typing as npt
from mpl_toolkits.mplot3d import axes3d
from numpy import int_, object_
from numpy.typing import NDArray
from test_utilities import test
from util import print_hex

COLORS = list(mcolors.CSS4_COLORS.keys())

<link href="style.css" rel="stylesheet"></link>
<article class="day-desc"><h2>--- PartI ---</h2</article>


In [ ]:
tests = [
    {
        "name": "Example",
        "s": """
            3   4
            4   3
            2   5
            1   3
            3   9
            3   3
        """,
        "expected": 11,
    },
]


@test(tests=tests)
def test_part_I(s: str) -> int:
    return -1


Test Example passed, for partI.
Success


In [ ]:
with open("../input/day1.txt") as f:
    puzzle = f.read()

# print(f"Part I: {total_distance(puzzle)}")

FileNotFoundError: [Errno 2] No such file or directory: '../input/day1.txt'

<link href="style.css" rel="stylesheet"></link>
<main>

<p>Your puzzle answer was <code>1722302</code>.</p><p class="day-success">The first half of this puzzle is complete! It provides one gold star: *</p>


<link href="style.css" rel="stylesheet"></link>
<article class="day-desc"><h2 id="part2">--- Part Two ---</h2></article>



Test Example passed, for partII.
Success


<link href="style.css" rel="stylesheet"></link>


In [ ]:
# print(f"Part I: {simmularity_score(puzzle)}")

Part I: 20373490


<link href="style.css" rel="stylesheet"></link>
<main>

<p>Your puzzle answer was <code>20373490</code>.</p><p class="day-success">Both parts of this puzzle are complete! They provide two gold stars: **</p>

</main>
